benchmarks/segarray_setops.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import time, argparse
+import numpy as np
+import arkouda as ak
+
+OPS = ('intersect_ma', 'union_ma', 'setdiff_ma', 'setxor_ma')
+TYPES = ('int64', 'uint64',)
+
+
+def time_ak_setops(N_per_locale, trials, dtype, seed):
+    print(">>> arkouda segarray {} setops".format(dtype))
+    cfg = ak.get_config()
+    N = N_per_locale * cfg["numLocales"]
+    print("numLocales = {}, N = {:,}".format(cfg["numLocales"], N))
+    if dtype == 'int64':
+        a = ak.randint(0, 2 ** 32, N, seed=seed)
+        b = ak.randint(0, 2 ** 32, N, seed=seed)
+        seg_a = ak.SegArray(ak.array([0, len(a)]), ak.concatenate([a, b]))
+        c = ak.randint(0, 2 ** 32, N, seed=seed)
+        d = ak.randint(0, 2 ** 32, N, seed=seed)
+        seg_b = ak.SegArray(ak.array([0, len(c)]), ak.concatenate([c, d]))
+    elif dtype == 'uint64':
+        a = ak.randint(0, 2 ** 32, N, seed=seed, dtype=ak.uint64)
+        b = ak.randint(0, 2 ** 32, N, seed=seed, dtype=ak.uint64)
+        seg_a = ak.SegArray(ak.array([0, len(a)]), ak.concatenate[[a, b]])
+        c = ak.randint(0, 2 ** 32, N, seed=seed, dtype=ak.uint64)
+        d = ak.randint(0, 2 ** 32, N, seed=seed, dtype=ak.uint64)
+        seg_b = ak.SegArray(ak.array([0, len(c)]), ak.concatenate([c, d]))
+
+    print(OPS)
+    timings = {op: [] for op in OPS}
+    results = {}
+    for i in range(trials):
+        for op in timings.keys():
+            fxn = getattr(seg_a, op)
+            start = time.time()
+            r = fxn(seg_b)
+            end = time.time()
+            timings[op].append(end - start)
+            results[op] = r
+    tavg = {op: sum(t) / trials for op, t in timings.items()}
+
+    for op, t in tavg.items():
+        print("  {} Average time = {:.4f} sec".format(op, t))
+        # bytes_per_sec = (a.size * a.itemsize * 2) / t
+        # print("  {} Average rate = {:.2f} GiB/sec".format(op, bytes_per_sec / 2 ** 30))
```

Comment:
was the average rate code meant to be left commented?

Reply:
Nope, uncommented it